src/dc_receive_imf.rs

```diff
@@ -991,7 +991,8 @@ INSERT INTO msgs
             }
         }
 
-        let mime_modified = save_mime_modified && !part.msg.is_empty();
+        let mime_modified =
+            save_mime_modified && !(part.msg.is_empty() && part.param.get(Param::Quote).is_none());
```

Comment:
@r10s do you remember why you introduced the `&& !part.msg.is_empty()` back then?

Because, I also thought about removing it, but it probably had a reason

Reply:
it is in there because if the message has more than one part, usually only one part contains text and we want to save that part (remember, the whole condition is checked in a look iterating over the message parts)